File: Algorithm_toolbox/ week 3 - greedy/fractional_knapsack/FractionalKnapsack.java
import java.util.Scanner;

public class FractionalKnapsack {
  private static double getOptimalValue(int n, int capacity, int[] values, int[] weights) {
    double value = 0;
    //write your code here

    while (n != 0)
    {
      int maxIndex = 0;
      for (int i=0 ; i<weights.length ; i++)
      {
        if (((double)values[i]/(double)weights[i] >= (double)values[maxIndex]/(double)weights[maxIndex]) && weights[i] > 0)
          maxIndex = i;
      }

      int a = weights[maxIndex] < capacity ? weights[maxIndex] : capacity;
      value = value + a*((double)values[maxIndex]/weights[maxIndex]);

      weights[maxIndex] -= a;
      //values[maxIndex] -= a;
      capacity -= a;
      n--;
    }
    return value;
  }

    public static void main(String args[]) {
      Scanner scanner = new Scanner(System.in);
      int n = scanner.nextInt();
      int capacity = scanner.nextInt();
      int[] values = new int[n];
      int[] weights = new int[n];
      for (int i = 0; i < n; i++) {
        values[i] = scanner.nextInt();
        weights[i] = scanner.nextInt();
      }
      System.out.println(getOptimalValue(n, capacity, values, weights));
    }
  } 

File: Algorithm_toolbox/ week 3 - greedy/fractional_knapsack/fractional_knapsack.cpp
#include <iostream>
#include <vector>

using std::vector;
using namespace std;

double get_optimal_value(int capacity, vector<int> weights, vector<int> values)
{

  cout<< "as as" ;
  double value = 0;

  while (capacity != 0)
  {
    int maxIndex = -1;
    for (int i=0 ; i<weights.size() ; i++)
    {
      if (((float)values[i]/(float)weights[i] > (float)values[maxIndex]/(float)weights[maxIndex]) && weights[i] > 0)
        maxIndex = i;
    }

    std::cout<<maxIndex;
    int a = weights[maxIndex] < capacity ? weights[maxIndex] : capacity;
    value = value + a*(values[maxIndex]/weights[maxIndex]);

    weights[maxIndex] -= a;
    //values[maxIndex] -= a;
    capacity -= a;


  }

  // write your code here

  return value;
}

int main() {
  int n;
  int capacity;

  std::cin >> n >> capacity;
  vector<int> values(n);
  vector<int> weights(n);

  for (int i = 0; i < n; i++) {

    std::cin >> values[i] >> weights[i];
  }


  cout<< "as as" ;
  double optimal_value = get_optimal_value(capacity, weights, values);

  std::cout << optimal_value << std::endl;
  return 0;
}
